provers/cairo/src/air.rs

```diff
@@ -160,12 +158,60 @@ pub const MEM_P_TRACE_OFFSET: usize = 17;
 pub const MEM_A_TRACE_OFFSET: usize = 19;
 
 #[derive(Debug, Clone, PartialEq, Eq, Hash, serde::Serialize, serde::Deserialize)]
-pub enum MemorySegment {
+pub enum SegmentName {
     RangeCheck,
     Output,
+    Program,
+    Execution,
+    Ecdsa,
+    Pedersen,
+}
+
+impl From<&str> for SegmentName {
+    fn from(value: &str) -> Self {
+        match value {
+            "range_check" => SegmentName::RangeCheck,
+            "output" => SegmentName::Output,
+            "program" => SegmentName::Program,
+            "execution" => SegmentName::Execution,
+            "ecdsa" => SegmentName::Ecdsa,
+            "pedersen" => SegmentName::Pedersen,
+            n => panic!("Invalid segment name {n}"),
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, serde::Serialize, serde::Deserialize)]
+pub struct Segment {
+    pub begin_addr: usize,
+    pub stop_ptr: usize,
+}
+
+impl Segment {
+    pub fn new(begin_addr: u64, stop_ptr: u64) -> Self {
+        let begin_addr = begin_addr.try_into().unwrap();
+        let stop_ptr = stop_ptr.try_into().unwrap();
+
+        Self {
+            begin_addr,
+            stop_ptr,
+        }
+    }
+    pub fn segment_size(&self) -> Option<usize> {
+        self.stop_ptr.checked_sub(self.begin_addr - 1)
```

Comment:
Also, for the first address (code segment, offset `0`) this will panic: `self.begin_addr - 1` is negative, which is out of range for `usize`. Maybe you want to add `1` after the subtraction instead?